kphp-inspector/src/file_searcher.js

```diff
@@ -38,42 +39,159 @@ function doesFileContainFunction(fullFileName) {
+
+    const searchPrefix = true;
+    const files = findFunctionFilesWithNameLike(filenameWithDot, searchPrefix);
+    utils.debug(`Found files: ${files.length === 0 ? 'empty' : files.join(', ')}`);
+
+    return files.filter(filename => doesFileContainFunction(filename));
   }
-  let longestQLen = Math.max(...qParts.map(p => p.length));
-  let longestQ = qParts.find(p => p.length === longestQLen);
 
-  /** @var {string[]} */
-  let output = child_process.execSync(`find ${env.RUN_ARGV.KPHP_COMPILED_ROOT} -iname "*${longestQ}*"`).toString().split('\n');
-  // strip out duplicates and strange files: leave only .cpp and .h, also filter out .h files of classes
-  output = output.filter(fn =>
-    fn.endsWith('.cpp') || (fn.endsWith('.h') && !output.includes(fn.substr(0, fn.length - 2) + '.cpp') && !fn.includes('cl/C@')));
+  let [queryParts, longestPart] = queryToParts(query);
+  const files = findFunctionFilesWithNameLike(longestPart);
+  utils.debug(`Found files: ${files.length === 0 ? 'empty' : files.join(', ')}`);
+
+  return files.filter(filename => doesFileSatisfyQ(filename, queryParts) && doesFileContainFunction(filename));
+}
+
+/**
+ * @param {string} query Search string from console
+ * @return {string[]}    Full file names, to be passed to parseCppClass()
+ */
+function performSearchForClass(query) {
+  const isFQN = query.startsWith('\\');
+  utils.debug(`Start search class by query: ${query}`);
 
-  return output.filter(fn => doesFileSatisfyQ(fn, qParts) && doesFileContainFunction(fn));
+  if (isFQN) {
+    const filename = convertFqnToKphpStyleCppFileName(query);
+    const filenameWithDot = filename + '\\.';
+    utils.debug(`Filename for search: ${filenameWithDot}`);
+
+    const files = findClassFilesWithNameLike(filenameWithDot);
+    utils.debug(`Found files: ${files.length === 0 ? 'empty' : files.join(', ')}`);
+
+    return files.filter(filename => filename.endsWith('.h'));
+  }
+
+  let [queryParts, longestPart] = queryToParts(query);
+  let files = findClassFilesWithNameLike(longestPart);
+  utils.debug(`Found files: ${files.length === 0 ? 'empty' : files.join(', ')}`);
+
+  return files.filter(filename => filename.endsWith('.h') && doesFileSatisfyQ(filename, queryParts));
 }
 
 /**
- * @param {string} q Search string from console
- * @return {string[]} Full file names, to be passed to parseCppClass()
+ * Convert search query into parts and looking for the longest.
+ * For example
+ *   "ClassName::method" -> parts: ['classname','method'], longest: 'classname'
+ *   "VK Feed something" -> parts: ['vk','feed','something'], longest: 'something'
+ *
+ * Returns the parts in first element and the longest part in second.
+ *
+ * @param {string} query
+ * @returns {[string[], string]}
  */
-function performSearchForClass(q) {
-  let qParts = splitUserSearchStr(q);
-  if (!qParts.length) {
-    return [];
+function queryToParts(query) {
+  let queryParts = splitUserSearchStr(query);
```

Comment:
Now, `splitUserSearchStr()` is being used only here and therefore seems a cognitive complication.
What do you think of inlining it here?

Reply:
Fixed.